pyramid/scripts/pserve.py

```diff
@@ -887,11 +889,28 @@ def check_reload(self):
                 continue
             if filename.endswith('.pyc') and os.path.exists(filename[:-1]):
                 mtime = max(os.stat(filename[:-1]).st_mtime, mtime)
+                pyc = True
+            else:
+                pyc = False
             if filename not in self.module_mtimes:
                 self.module_mtimes[filename] = mtime
             elif self.module_mtimes[filename] < mtime:
-                print("%s changed; reloading..." % filename)
-                return False
+                if pyc:
+                    filename = filename[:-1]
+                # check if a file has syntax errors. If so, track it until it's fixed
+                for filename_promise in self.syntax_error_files + [filename]:
+                    try:
+                        py_compile.compile(filename_promise, doraise=True)
+                    except py_compile.PyCompileError:
+                        print("%s has a SyntaxError; NOT reloading." % filename_promise)
```

Comment:
At the head of this file, logging is set up. Shouldn't we use that logger instead of `print`?